### ISSUE TITLE
[FEATURE] - MazDataTable

### ISSUE THREAD

**Issue by caio-emidio** [LouisMazel/maz-ui#777]
**Is your feature request related to a problem? Please describe.**
In my project (ERP) i have the use of many tables. Maybe have a MazDataTable would be a great addition to the UI. 

**Describe the solution you'd like**
Possibility to: 
 - Input Headers
 - Data
 - Custom Data
 - Sortable

**Describe alternatives you've considered**
its 100% possible to do separate. Its just to grow the UI.

**Additional context**
If need another Add Context. I will be happy to help

**Comment by LouisMazel** [comment]
Hey, 

Yes, it's a good idea and I want to do this for a long time! 
Do you have examples or specs to share me?

Thanks

**Comment by caio-emidio** [comment]
```vue
<template>
  <div class="overflow-x-auto">
    <table class="min-w-full divide-y divide-gray-200">
      <thead>
        <tr>
          <th
            v-for="column in headers"
            :key="column.key"
            @click="sortBy(column)"
            :class="{ sortable: column.sorted }"
            class=" py-3 bg-gray-50 dark:bg-black/70 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
          >
            {{ column.label }}
            <span v-if="column.sorted && sortByColumn === column.key">
              {{ sortDirection === "asc" ? "▲" : "▼" }}
            </span>
          </th>
        </tr>
      </thead>
      <tbody class="bg-white dark:bg-black divide-y divide-gray-200">
        <tr v-for="item in sortedItems" :key="item.id">
          <td v-for="column in headers" :key="column.key" class="px-6 py-4 whitespace-nowrap">
            <slot :name="column.key" :item="item">{{ item[column.key] }}</slot>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</template>

<script>
export default {
  props: {
    headers: Array,
    items: Array,
  },
  data() {
    return {
      sortByColumn: null,
      sortDirection: "asc",
    };
  },
  computed: {
    sortedItems() {
      if (this.sortByColumn) {
        const direction = this.sortDirection === "asc" ? 1 : -1;
        return this.items.slice().sort((a, b) => {
          return a[this.sortByColumn] < b[this.sortByColumn]
            ? -direction
            : direction;
        });
      }
      return this.items;
    },
  },
  methods: {
    sortBy(column) {
      if (column.sorted) {
        if (this.sortByColumn === column.key) {
          this.sortDirection = this.sortDirection === "asc" ? "desc" : "asc";
        } else {
          this.sortByColumn = column.key;
          this.sortDirection = "asc";
        }
      }
    },
  },
};
</script>
```

Example of use:

```vue
const headers = ref([
  { key: "name", label: "Name", sorted: true },
  { key: "relationedAccount", label: "Relationed Account", sorted: true },
  { key: "active", label: "Active", sorted: true },
  { key: "actions", label: "Actions", sorted: false },
]);
```

**Comment by caio-emidio** [comment]
Look if that could help you :)

**Comment by caio-emidio** [comment]
i have other features then i would like to implement on UI. 
SideBar, Navbar and etc..

**Comment by LouisMazel** [comment]
The sidebar already exists, check out the MazDrawer component 

For the navbar, I don't think it will ever exist. It’s really simple to do and depends on each project. I only make useful and complex components from now on

**Comment by LouisMazel** [comment]
@caio-emidio 

The component `MazTable` will be released soon, follow this PR: https://github.com/LouisMazel/maz-ui/pull/829

I have to document it before released and then all is good :)

**Comment by caio-emidio** [comment]
Thanks for that! 
Happy X-MAS

**Comment by LouisMazel** [comment]
The component "MazTable" is available in v3.30.0

* **Release Note** - https://github.com/LouisMazel/maz-ui/releases/tag/v3.30.0
* **Documentation** - https://louismazel.github.io/maz-ui-3/components/maz-table

Don't hesitate to ask me if you have any questions!

Merry x-mas @caio-emidio :)
